### PR TITLE
179816806 create post

### PR DESCRIPTION
**Pull request by mmyrtice**
my tasks
[As User I can see my posts](https://www.pivotaltracker.com/story/show/179816806)
[As User I can make a Post](https://www.pivotaltracker.com/story/show/179816806)
+ comments

### CODE REVIEW

File: app/helpers/comments_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (AidarMingazov) [inline app/helpers/comments_helper.rb]: pls remove extra files

File: app/helpers/posts_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
```
Comment (AidarMingazov) [inline app/helpers/posts_helper.rb]: pls remove extra files

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Comment < ApplicationRecord
```
Comment (AidarMingazov) [inline app/models/comment.rb]: pls remove extra

File: app/models/concerns/visible.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  class_methods do
+    def public_count
+      where(status: 'public').count
+    end
+  end
+
+  def archived?
+    status == 'archived'
+  end
+end
```
Comment (AidarMingazov) [inline app/models/concerns/visible.rb]: pls add enter

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+   include Visible
+
+   has_many :comments, dependent: :destroy
+
```
Comment (AidarMingazov) [inline app/models/post.rb:7]: pls user Enum inside `include Visible`

File: app/views/comments/_comment.html.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<% unless comment.archived? %>
```
Comment (AidarMingazov) [inline app/views/comments/_comment.html.erb]: extra file

File: app/views/posts/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<h1>Edit Post</h1>
+
+<%= render "form", post: @post %>
```
Comment (AidarMingazov) [inline app/views/posts/edit.html.erb]: end of file

File: app/views/posts/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  <div>
+    <%= form.label :status %><br>
+    <%= form.select :status, ['public', 'private', 'archived'], selected: 'public' %>
+  </div>
+
+  <div>
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (AidarMingazov) [inline app/views/posts/_form.html.erb]: end of file

File: app/views/posts/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<h1>New Post</h1>
+
+<%= render "form", post: @post %>
```
Comment (AidarMingazov) [inline app/views/posts/new.html.erb]: end of file

File: config/routes.rb [hunk 10]
```diff
@@ -2,5 +2,9 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  root "posts#index"
+
+  resources :posts do
+    resources :comments
```
Comment (AidarMingazov) [inline config/routes.rb]: extra

File: db/migrate/20220609123227_create_posts.rb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :body
```
Comment (AidarMingazov) [inline db/migrate/20220609123227_create_posts.rb:5]: add `status`

File: db/migrate/20220609125616_create_comments.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
```
Comment (AidarMingazov) [inline db/migrate/20220609125616_create_comments.rb]: extra

File: db/migrate/20220609132936_add_status_to_comments.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToComments < ActiveRecord::Migration[7.0]
```
Comment (AidarMingazov) [inline db/migrate/20220609132936_add_status_to_comments.rb]: extra

File: test/controllers/comments_controller_test.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (AidarMingazov) [inline test/controllers/comments_controller_test.rb]: extra

File: test/controllers/posts_controller_test.rb [hunk 15]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (AidarMingazov) [inline test/controllers/posts_controller_test.rb]: extra

File: test/fixtures/comments.yml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (AidarMingazov) [inline test/fixtures/comments.yml]: extra

File: test/fixtures/posts.yml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (AidarMingazov) [inline test/fixtures/posts.yml]: extra

File: test/models/comment_test.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (AidarMingazov) [inline test/models/comment_test.rb]: extra

File: test/models/post_test.rb [hunk 19]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (AidarMingazov) [inline test/models/post_test.rb]: extra

**Review by AidarMingazov — CHANGES_REQUESTED** [review]

File: db/migrate/20220609123227_create_posts.rb (hunk 11 above)
Comment (BoutPure) [reply db/migrate/20220609123227_create_posts.rb:5]: Added in separate migration

File: app/models/post.rb (hunk 5 above)
Comment (BoutPure) [reply app/models/post.rb:7]: I had put enum to posts without Visible module for now because I can't make it work inside one

**Review by BoutPure — APPROVED** [review]
I fix previous review comments.
Delete comments from project.
Add end file newlines.
Swap post statuses string array to enum.
Approved to merge.

File: app/controllers/posts_controller.rb [hunk 20]
```diff
@@ -0,0 +1,49 @@
+class PostsController < ApplicationController
+
+  def index
+    @posts = Post.all
+  end
+
+  def show
+      @post = Post.find(params[:id])
```
Comment (AidarMingazov) [inline app/controllers/posts_controller.rb]: extra tab

File: app/controllers/posts_controller.rb [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    p @post
```
Comment (AidarMingazov) [inline app/controllers/posts_controller.rb]: what happen here ?

File: app/models/concerns/visible.rb [hunk 22]
```diff
@@ -0,0 +1,19 @@
+module Visible
```
Comment (AidarMingazov) [inline app/models/concerns/visible.rb]: extra file

**Review by AidarMingazov — CHANGES_REQUESTED** [review]

**Pull request merged**